wpilib/src/robot.rs

```diff
@@ -89,22 +94,28 @@ fn loop_func<T: IterativeRobot>(
 ///
 /// It is recommended to use `start_timed` instead,
 /// which guarantees a more regular period of execution.
-pub fn start_iterative<T: IterativeRobot>(robot: &mut T, ds: &DriverStation) {
+pub fn start_iterative<T: IterativeRobot>() -> ! {
+    let base = RobotBase::new().expect("Could not initialize HAL");
+    let ds = base.make_ds();
+
+    println!("\n********** Robot program starting **********\n");
+
+    let mut robot = T::new(&ds);
     let mut last_mode: Option<RobotState> = None;
 
     usage::report(
         usage::resource_types::Framework,
         usage::instances::kFramework_Iterative,
     );
 
-    println!("\n********** Robot program starting **********\n");
+    // Tell the DS that the robot is ready to be enabled
     unsafe { HAL_ObserveUserProgramStarting() }
```

Comment:
nit: could/should be `RobotBase::start_competition()` That also prints the `*** program starting ***` message.